.devcontainer/dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM ubuntu:22.04
+ARG USERNAME=sonata
+# Install nix, git and cmake (required to build sonata-system software,
+# temporary addition to this dockerfile until we have software build setup in
+# sonata-software).
+RUN apt update -y
+RUN apt install curl git cmake -y
+
+# Setup user
+RUN useradd -m $USERNAME \
+    # Add sudo support.
+    && apt-get install -y sudo \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
+    && chmod 0440 /etc/sudoers.d/$USERNAME
+
+# Switch to user
+USER $USERNAME
+
+# Install nix
+RUN curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install linux \
+  --extra-conf "sandbox = false" \
+  --init none \
+  --no-confirm
+
+RUN sudo chown -R $USERNAME /nix
+
+ENV PATH="${PATH}:/nix/var/nix/profiles/default/bin"
+
+# Setup lowRISC nix cache
+RUN sudo bash -c 'echo "substituters = https://cache.nixos.org https://nix-cache.lowrisc.org/public/" >> /etc/nix/nix.conf'
+RUN sudo bash -c 'echo "trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=  nix-cache.lowrisc.org-public-1:O6JLD0yXzaJDPiQW1meVu32JIDViuaPtGDfjlOopU7o=" >> /etc/nix/nix.conf'
+RUN sudo bash -c "echo trusted-users = $USERNAME >> /etc/nix/nix.conf"
+RUN sudo bash -c "echo 'bash-prompt = \[\e[1m\]Sonata \[\e[0m\]\w>\040' >> /etc/nix/nix.conf"
+RUN sudo bash -c "echo 'bash-prompt-prefix =' >> /etc/nix/nix.conf"
+
+# Stop nix from asking about whether we trust cache settings
+RUN mkdir -p /home/$USERNAME/.local/share/nix
+RUN echo '{"extra-substituters":{"https://nix-cache.lowrisc.org/public/":true},"extra-trusted-public-keys":{"nix-cache.lowrisc.org-public-1:O6JLD0yXzaJDPiQW1meVu32JIDViuaPtGDfjlOopU7o=":true}}' > /home/$USERNAME/.local/share/nix/trusted-settings.json
```

Comment:
I put it in because when I was running `nix develop ...` it asked me to confirm various things then wrote this `trusted-settings.json` file.

You're right this is maybe surprising but I just  wanted to get something working for now.